torch/csrc/jit/codegen/cuda/lower_loops.cpp

```diff
@@ -108,12 +108,11 @@ void LoopNestGenerator::handle(const Expr* expr) {
   // Fill the entire loop structure by Looking at each axis
   // individually in out's domain
   for (size_t out_i = 0; out_i < out_tv->nDims(); out_i++) {
-    auto out_id = out_tv->axis(out_i);
-    // If out_id is derived from trivial reductions and its root axes
-    // are also all the case, it's safe to skip this axis.
-    if (gpu_lower->trivialReductionInfo().isDerivedFromRoot(out_id)) {
-      continue;
-    }
+    // Note: It is not safe to skip trivial reduction axes as they could be
+    // inlined with other tensor views. This happens in
+    // NVFuserTest.FusionBNRepro_CUDA as of this commit on norm_hack_2_rebased
+    // branch
+
```

Comment:
This is fine and should be done so if anything breaks. However, if it is actually safe to skip, this change would mean we would generate unnecessary loops.

For example, here are excerpts from the `FusionTrivialReduction1` test with and without this change.

Current:
```
__global__ void kernel1(Tensor<float, 1> T0, Tensor<float, 1> T2, Tensor<float, 1> T8, Tensor<float, 1> T11, Tensor<float, 3> T3) {
  #pragma unroll 1
  for(size_t ki136 = 0; ki136 < T0.size[0]; ++ki136) {
    float T9[1];
    T9[0]
       = T0[(ki136 * T0.stride[0])];
    float T10[1];
    T10[0]
       = T9[0];
    T11[(ki136 * 1)]
       = T10[0];
  }
``` 

With this change:
```
__global__ void kernel1(Tensor<float, 1> T0, Tensor<float, 1> T2, Tensor<float, 1> T8, Tensor<float, 1> T11, Tensor<float, 3> T3) {
  #pragma unroll 1
  for(size_t ki142 = 0; ki142 < T0.size[0]; ++ki142) {
    float T9[1];
    T9[0]
       = T0[(ki142 * T0.stride[0])];
    float T10[1];
    #pragma unroll 1
    for(size_t ki150 = 0; ki150 < (ceilDiv(1, 4)); ++ki150) {
      #pragma unroll 1
      for(size_t ki152 = 0; ki152 < 4; ++ki152) {
        T10[0]
           = T9[0];
      }
    }
    #pragma unroll 1
    for(size_t ki144 = 0; ki144 < (ceilDiv(1, 5)); ++ki144) {
      #pragma unroll 1
      for(size_t ki146 = 0; ki146 < 5; ++ki146) {
        T11[(ki142 * 1)]
           = T10[0];
      }
    }
  }
```

The compiler should be able to optimize away the trivial loops in this case, but that might depend on other factors.

It should be also possible to skip loop generation only when no other tensor is computed-at. I'll create an issue to revisit this if desired for performance reasons.